File: 2384-root-equals-sum-of-children/2384-root-equals-sum-of-children.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void valid(TreeNode* root, bool& res) {
        if (root == nullptr) {
            res = false;
            return;
        }
        if (!root->left || !root->right) {
            res = false;
            return;
        }
        if (root->val == root->left->val + root->right->val) {
            res = true;
            return;
        }
        valid(root->left, res);
        valid(root->right, res);
    }
    bool checkTree(TreeNode* root) {
        bool res = false;
        valid(root, res);
        return res;
    }
};

File: 2384-root-equals-sum-of-children/2384-root-equals-sum-of-children.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    

    def checkTree(self, root: Optional[TreeNode]) -> bool:
        res = False  
        def valid(root: Optional[TreeNode]) -> None:
            nonlocal res
            if not root:
                res = False
                return
            if not root.left or not root.right:
                res = False
                return
            if root.val == root.left.val + root.right.val:
                res = True
                return
            valid(root.left)
            valid(root.right)
        valid(root)
        return res